aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3802,4 +3802,226 @@ Tensor _det_lu_based_helper_backward_helper(
+  //
+  // # Here we undo the conjugations / transposes on out_f if needed
+  //
+  // if out_f not same out:
+  //   copy out_f into out
+  // return out
+  //
+  // Note: The logic for the negative bit is the same as that for the conjugate bit
+  //
+  // Note: [Cloning A] If we are careful when allocating B when it needs to be allocated at the
+  // beginning of the algorithm, it is possible to always elide the copy of A here.
+  // Via this trick, the algorithm will copy at most one of A or B (never both) whenever A
+  // and B are F-ready and not A.is_neg() (which happens almost always in practice).
+  // When called as f(A, B, out=B) in most practical cases it'll perform no copies.
+
+  const bool avoid_copy_A = A_.transpose(-2, -1).is_contiguous() && A_.is_conj();
+  if (avoid_copy_A) {
+    // See Note: [Cloning A]
+    at::native::resize_output(out, B_.sizes());
+  }
+  else {
+    // poorman's reimplementation of resize_output with result F-contig
+    if (resize_output_check(out, B_.sizes())) {
+      out.resize_(B_.transpose(-2, -1).sizes(), MemoryFormat::Contiguous);
+      out.transpose_(-2, -1);  // make 'out' have Fortran contiguous memory layout
+    }
+  }
+  // Invariant: out has the right size, so we'll be able to copy into it later on
+
+  Tensor out_f; // the out that will go into fortran
+  // We use C10_LIKELY mostly for documentation as it helps following what's the most likely path
+  if C10_LIKELY (is_row_or_column_contiguous(out)) {
+    out_f = out;
+    if C10_LIKELY (!out.is_same(B_)) {
+      out_f.copy_(B_);
+    }
+  } else {
+    if (avoid_copy_A) {
+      // See Note: [Cloning A]
+      out_f = B_.clone(at::MemoryFormat::Contiguous);
+    }
+    else {
+      out_f = cloneBatchedColumnMajor(B_);
+    }
+  }
+  // Invariant: out_f F-ready and has B copied into it
+
+  // out_f is F-transposed
+  bool transpose_A = false;
+  bool transpose_out_f = false;
+  if (out_f.stride(-1) == 1) {
+    left = !left;
+    transpose_A = true;
+    transpose_out_f = true;
+    out_f.transpose_(-2 ,-1);
+  }
+
+  // No need to conjugate anything if out_f is conj as AX = conj(B) <=> conj(A)conj(X) = B
+  // and X = B after the algortihm. We just anotate that A is conjugated later on
+  // The solution will be written into out_f, so it'll be conjugated already
+
+  Tensor A_f;  // The A that will go into fortran
+  bool A_is_conj = A_.is_conj() != out_f.is_conj();
+  bool A_is_neg = A_.is_neg() != out_f.is_neg();
+  bool A_is_f_contig = (A_.stride(-1) == 1) == transpose_A;
+  if C10_LIKELY (is_row_or_column_contiguous(A_) && !((A_is_conj && A_is_f_contig) || A_is_neg)) {
+    A_f = A_;
+  } else {
+    // We choose C-contig rather than F-contig because it has better memory access in solvers
+    // We resolve the conj as well
+    A_f = A_.clone(at::MemoryFormat::Contiguous);
+    A_is_f_contig = transpose_A;
+    A_is_conj = false;
+    A_is_neg = false;
```

Comment:
The negative bit is not explained in the comments above. I was surprised to discover it, actually :)

Reply:
Yeah, me too, I learnt about it when a test from the OpInfo failed :D 
It's what you get when you do `torch.rand(3, 3, dtype=torch.complex64).conj().imag`.
I think I wrote a small note saying that it's handled in the same way as the conjugate bit, as multiplying by `-1` has the same properties as conjugation.

Reply:
Yes, this note was written, but I thought of a negative bit related to number representation, not in the context of Tensors :)